internal/controller/evictor.go

```diff
@@ -191,39 +191,52 @@ func (e *evictor) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(e)
 }
 
-// getProvionedImagePullSecret gets an image pull secret provisioned for a ServiceAccount.
-// It returns an empty string if there is no image pull secret provisioned.
-func (e *evictor) getProvisionedImagePullSecret(
+// getProvionedImagePullSecrets gets image pull secrets provisioned for a ServiceAccount.
+// It returns an empty slice if there are no image pull secrets provisioned.
+func (e *evictor) getProvisionedImagePullSecrets(
```

Comment:
Image pull secrets provisioned for a ServiceAccount have a label, so I think we can list the secrets using the label.